### PR TITLE
feature/JSSNMSHDD-2810 Avoid DB query for page size when less than one page exists

### PR DESCRIPTION
**Pull request by tnotheis**

### CODE REVIEW

File: Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs [hunk 1]
```diff
@@ -29,15 +29,15 @@ public async Task<GetDatawalletModificationsResult> GetDatawalletModifications(I
...
-        var totalNumberOfItems = await query.CountAsync();
-
+        
         query = query
             .OrderBy(m => m.Index)
             .Paged(paginationFilter);
 
         var items = await query.ToListAsync();
 
+        var totalNumberOfItems = items.Count < paginationFilter.PageSize ? items.Count : await query.CountAsync();
```
Comment (stnmtz) [inline Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs:39]: <= ?

File: Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs [hunk 2]
```diff
@@ -156,15 +156,15 @@ public async Task<(IEnumerable<ExternalEvent> firstPage, int totalRecords)> GetE
...
-
-        var events = await query
+        
+        var items = await query
             .OrderBy(d => d.Index)
             .Paged(paginationFilter)
             .ToListAsync(cancellationToken);
 
-        return (events, totalRecords);
+        var totalNumberOfItems = items.Count < paginationFilter.PageSize ? items.Count : await query.CountAsync(cancellationToken);
```
Comment (stnmtz) [inline Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs:165]: <= ?

**Review by stnmtz — CHANGES_REQUESTED** [review]

File: Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs (hunk 2 above)
Comment (jkoenig134) [reply Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs:165]: +1

File: Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs (hunk 1 above)
Comment (jkoenig134) [reply Synchronization.Infrastructure/Persistence/Database/ApplicationDbContext.cs:39]: +1

**Review by jkoenig134 — APPROVED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Review by stnmtz — APPROVED** [review]

**Pull request merged**